provider/alibabacloud/alibaba_cloud.go

```diff
@@ -528,9 +528,13 @@ func (p *AlibabaCloudProvider) createRecord(endpoint *endpoint.Endpoint, target
 }
 
 func (p *AlibabaCloudProvider) createRecords(endpoints []*endpoint.Endpoint) error {
+	hostedZoneDomains, err := p.getDomainList()
```

Comment:
Consider doing this in the caller, so that you get the domain list once per update, not three times.